build/build.go

```diff
@@ -1235,6 +1238,33 @@ func LoadInputs(ctx context.Context, d driver.Driver, inp Inputs, pw progress.Wr
 			target.FrontendAttrs["context:"+k] = v.Path
 			continue
 		}
+
+		// handle OCI layout
+		if strings.HasPrefix(v.Path, "oci-layout://") {
+			pathAlone := strings.TrimPrefix(v.Path, "oci-layout://")
+			// need to add "dummy.io", because reference.Parse*() funcs require we do not start with a '/'
+			named, err := reference.ParseNormalizedNamed(fmt.Sprintf("dummy.io%s", pathAlone))
```

Comment:
Rather than this hack just split by `@` and then `digest.Parse` maybe.

Reply:
One ugly hack vs another? I was _so_ disappointed than `reference.Parse` returns a thing that does not have component parts that can be looked up, like `url.Parse()` does.